liquibase-couchbase/src/main/java/liquibase/ext/couchbase/validator/JsonChangelogValidator.java

```diff
@@ -21,8 +21,18 @@
 public class JsonChangelogValidator implements SingletonObject {
 
     private final Logger log = Scope.getCurrentScope().getLog(getClass());
-    private final ObjectMapper objectMapper = new ObjectMapper();
-    private final JsonSchemaFactory schemaFactory = JsonSchemaFactory.getInstance(SpecVersion.VersionFlag.V201909);
+    private final ObjectMapper objectMapper;
+    private final JsonSchemaFactory schemaFactory;
+
+    public JsonChangelogValidator() {
+        this.objectMapper = new ObjectMapper();
+        this.schemaFactory = JsonSchemaFactory.getInstance(SpecVersion.VersionFlag.V201909);
+    }
+
+    public JsonChangelogValidator(ObjectMapper objectMapper, JsonSchemaFactory schemaFactory) {
```

Comment:
I haven't found any usage of this constructor. Is it required?

Reply:
It's a tricky one. I need to mock ObjectMapper and Factory objects for unit tests. But we don't have appropraite constructors or setters for this. Options which I have:

1. set object in tests using reflection
2. remove final from field and add setters
3. add new constructor
4. maybe we have other options and I missed it

I know that adding code specially for test is bad idea and I agree to find another option. What do you think?

Reply:
@DaimonCool  
No, code exclusively for testing **is not** a bad idea.
You can make this constructor `package-private` and for clarity add annotation `@VisibleForTesting`
Go ahead pls with new constructor, but ensure that test package is the same as original class

Reply:
@DaimonCool 
I agree that if small refactoring could help to help to write better unit test - that should be done.
But my comment is related to another condition. Please, take a look on that screenshot https://prnt.sc/659XBcwzYwQj

Reply:
Thanks for comments. Done